src/field.h

```diff
@@ -9,6 +9,8 @@
 #include "color.h"
 #include "field_type.h"
 
+class effect;
```

Comment:
I think this isn't needed here since `field_effect()` is already returning by value?

Reply:
this is needed, what's not needed is including the whole file.